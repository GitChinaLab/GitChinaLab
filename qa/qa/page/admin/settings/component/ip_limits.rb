# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class IpLimits < Page::Base
            view 'app/views/admin/application_settings/_ip_limits.html.haml' do
              element :throttle_unauthenticated_api_checkbox
              element :throttle_unauthenticated_web_checkbox
              element :throttle_authenticated_api_checkbox
              element :throttle_authenticated_web_checkbox
              element :save_changes_button
            end

            def enable_throttles
              check_element(:throttle_unauthenticated_api_checkbox, true)
              check_element(:throttle_unauthenticated_web_checkbox, true)
              check_element(:throttle_authenticated_api_checkbox, true)
              check_element(:throttle_authenticated_web_checkbox, true)
            end

            def save_settings
              click_element :save_changes_button
            end
          end
        end
      end
    end
  end
end
