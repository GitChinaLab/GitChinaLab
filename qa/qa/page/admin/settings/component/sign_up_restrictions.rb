# frozen_string_literal: true

module QA
  module Page
    module Admin
      module Settings
        module Component
          class SignUpRestrictions < Page::Base
            view 'app/assets/javascripts/pages/admin/application_settings/general/components/signup_form.vue' do
              element :require_admin_approval_after_user_signup_checkbox
              element :signup_enabled_checkbox
              element :save_changes_button
            end

            def require_admin_approval_after_user_signup
              click_element_coordinates(:require_admin_approval_after_user_signup_checkbox, visible: false)
              click_element(:save_changes_button)
            end

            def disable_signups
              click_element_coordinates(:signup_enabled_checkbox, visible: false)
              click_element(:save_changes_button)
            end
          end
        end
      end
    end
  end
end
