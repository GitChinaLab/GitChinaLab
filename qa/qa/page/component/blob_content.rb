# frozen_string_literal: true

module QA
  module Page
    module Component
      module BlobContent
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/assets/javascripts/blob/components/blob_header_filepath.vue' do
            element :file_title_content
          end

          base.view 'app/assets/javascripts/blob/components/blob_content.vue' do
            element :blob_viewer_file_content
          end

          base.view 'app/assets/javascripts/blob/components/blob_header_default_actions.vue' do
            element :default_actions_container
            element :copy_contents_button
          end

          base.view 'app/views/projects/blob/_header_content.html.haml' do
            element :file_name_content
          end

          base.view 'app/views/shared/_file_highlight.html.haml' do
            element :file_content
          end
        end

        def has_file?(name)
          has_file_name?(name)
        end

        def has_no_file?(name)
          has_no_file_name?(name)
        end

        def has_file_name?(file_name, file_number = nil)
          within_file_by_number(file_name_element, file_number) { has_text?(file_name) }
        end

        def has_no_file_name?(file_name)
          within_element(file_name_element) do
            has_no_text?(file_name)
          end
        end

        def has_file_content?(file_content, file_number = nil)
          within_file_by_number(file_content_element, file_number) { has_text?(file_content) }
        end

        def has_no_file_content?(file_content)
          within_element(file_content_element) do
            has_no_text?(file_content)
          end
        end

        def click_copy_file_contents(file_number = nil)
          within_file_by_number(:default_actions_container, file_number) { click_element(:copy_contents_button) }
        end

        private

        def file_content_element
          feature_flag_controlled_element(:refactor_blob_viewer, :blob_viewer_file_content, :file_content)
        end

        def file_name_element
          feature_flag_controlled_element(:refactor_blob_viewer, :file_title_content, :file_name_content)
        end

        def within_file_by_number(element, file_number)
          if file_number
            within_element_by_index(element, file_number - 1) { yield }
          else
            within_element(element) { yield }
          end
        end
      end
    end
  end
end
