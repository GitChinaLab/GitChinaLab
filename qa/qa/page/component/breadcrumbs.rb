# frozen_string_literal: true

module QA
  module Page
    module Component
      module Breadcrumbs
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/views/layouts/nav/_breadcrumbs.html.haml' do
            element :breadcrumb_links_content
          end
        end

        def has_breadcrumb?(text)
          has_element?(:breadcrumb_links_content, text: text)
        end
      end
    end
  end
end
