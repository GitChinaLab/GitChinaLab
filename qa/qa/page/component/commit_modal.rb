# frozen_string_literal: true

module QA
  module Page
    module Component
      class CommitModal < Page::Base
        view 'app/assets/javascripts/projects/commit/components/form_modal.vue' do
          element :submit_commit_button, required: true
        end
      end
    end
  end
end
