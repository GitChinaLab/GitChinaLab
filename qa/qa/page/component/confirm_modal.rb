# frozen_string_literal: true

module QA
  module Page
    module Component
      module ConfirmModal
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/views/shared/_confirm_modal.html.haml' do
            element :confirm_modal
            element :confirm_input
            element :confirm_button
          end
        end

        def fill_confirmation_text(text)
          fill_element :confirm_input, text
        end

        def wait_for_confirm_button_enabled
          wait_until(reload: false) do
            !find_element(:confirm_button).disabled?
          end
        end

        def confirm_transfer
          wait_for_confirm_button_enabled
          click_element :confirm_button
        end
      end
    end
  end
end
