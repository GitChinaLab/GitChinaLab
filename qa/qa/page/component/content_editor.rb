# frozen_string_literal: true

module QA
  module Page
    module Component
      module ContentEditor
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/assets/javascripts/content_editor/components/content_editor.vue' do
            element :content_editor_container
          end

          base.view 'app/assets/javascripts/content_editor/components/toolbar_text_style_dropdown.vue' do
            element :text_style_dropdown
            element :text_style_menu_item
          end

          base.view 'app/assets/javascripts/content_editor/components/toolbar_image_button.vue' do
            element :file_upload_field
          end
        end

        def add_heading(heading, text)
          within_element(:content_editor_container) do
            text_area.set(text)
            # wait for text style option to become active after typing
            has_active_element?(:text_style_dropdown, wait: 1)
            click_element(:text_style_dropdown)
            within_element(:text_style_dropdown) do
              click_element(:text_style_menu_item, text_style: heading)
            end
          end
        end

        def upload_image(image_path)
          within_element(:content_editor_container) do
            # add image on a new line
            text_area.send_keys(:return)
            find_element(:file_upload_field, visible: false).send_keys(image_path)
          end
        end

        private

        def text_area
          find('[contenteditable="true"]', visible: false)
        end
      end
    end
  end
end
