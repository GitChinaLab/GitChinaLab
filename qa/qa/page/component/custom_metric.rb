# frozen_string_literal: true

module QA
  module Page
    module Component
      module CustomMetric
        extend QA::Page::PageConcern

        def self.included(base)
          super

          base.view 'app/assets/javascripts/custom_metrics/components/custom_metrics_form_fields.vue' do
            element :custom_metric_prometheus_title_field
            element :custom_metric_prometheus_query_field
            element :custom_metric_prometheus_y_label_field
            element :custom_metric_prometheus_unit_label_field
            element :custom_metric_prometheus_legend_label_field
          end
        end

        def add_custom_metric
          fill_element :custom_metric_prometheus_title_field, 'HTTP Requests Total'
          fill_element :custom_metric_prometheus_query_field, 'rate(http_requests_total[5m])'
          fill_element :custom_metric_prometheus_y_label_field, 'Requests/second'
          fill_element :custom_metric_prometheus_unit_label_field, 'req/sec'
          fill_element :custom_metric_prometheus_legend_label_field, 'HTTP requests'

          save_changes
        end

        def save_changes
          click_button(class: 'btn-success')
        end

        def delete_custom_metric
          click_button(class: 'btn-danger')
          within('.modal-content') { click_button(class: 'btn-danger') }
        end

        def edit_custom_metric
          fill_element :custom_metric_prometheus_title_field, ''
          fill_element :custom_metric_prometheus_title_field, 'Throughput'

          save_changes
        end
      end
    end
  end
end
