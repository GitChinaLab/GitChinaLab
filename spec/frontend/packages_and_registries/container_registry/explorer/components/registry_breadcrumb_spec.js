import { mount } from '@vue/test-utils';

import component from '~/packages_and_registries/container_registry/explorer/components/registry_breadcrumb.vue';

describe('Registry Breadcrumb', () => {
  let wrapper;
  const nameGenerator = jest.fn();

  const routes = [
    { name: 'list', path: '/', meta: { nameGenerator, root: true } },
    { name: 'details', path: '/:id', meta: { nameGenerator } },
  ];

  const mountComponent = ($route) => {
    wrapper = mount(component, {
      mocks: {
        $route,
        $router: {
          options: {
            routes,
          },
        },
      },
    });
  };

  beforeEach(() => {
    nameGenerator.mockClear();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  describe('when is rootRoute', () => {
    beforeEach(() => {
      mountComponent(routes[0]);
    });

    it('renders', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('contains only a single router-link to list', () => {
      const links = wrapper.findAll('a');

      expect(links).toHaveLength(1);
      expect(links.at(0).attributes('href')).toBe('/');
    });

    it('the link text is calculated by nameGenerator', () => {
      expect(nameGenerator).toHaveBeenCalledTimes(1);
    });
  });

  describe('when is not rootRoute', () => {
    beforeEach(() => {
      mountComponent(routes[1]);
    });

    it('renders', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('contains two router-links to list and details', () => {
      const links = wrapper.findAll('a');

      expect(links).toHaveLength(2);
      expect(links.at(0).attributes('href')).toBe('/');
      expect(links.at(1).attributes('href')).toBe('#');
    });

    it('the link text is calculated by nameGenerator', () => {
      expect(nameGenerator).toHaveBeenCalledTimes(2);
    });
  });
});
