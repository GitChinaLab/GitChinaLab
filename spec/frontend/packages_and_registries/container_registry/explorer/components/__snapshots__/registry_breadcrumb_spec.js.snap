// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry Breadcrumb when is not rootRoute renders 1`] = `
<div
  class="gl-breadcrumbs"
>
   
  <ol
    class="breadcrumb gl-breadcrumb-list"
  >
    <li
      class="breadcrumb-item gl-breadcrumb-item"
    >
      <a
        class=""
        href="/"
        target="_self"
      >
        <span>
          
        </span>
         
        <span
          class="gl-breadcrumb-separator"
          data-testid="separator"
        >
          <svg
            aria-hidden="true"
            class="gl-icon s8"
            data-testid="angle-right-icon"
            role="img"
          >
            <use
              href="#angle-right"
            />
          </svg>
        </span>
      </a>
    </li>
    <li
      class="breadcrumb-item gl-breadcrumb-item"
    >
      <a
        class=""
        href="#"
        target="_self"
      >
        <span>
          
        </span>
         
        <!---->
      </a>
    </li>
  </ol>
</div>
`;

exports[`Registry Breadcrumb when is rootRoute renders 1`] = `
<div
  class="gl-breadcrumbs"
>
   
  <ol
    class="breadcrumb gl-breadcrumb-list"
  >
    <li
      class="breadcrumb-item gl-breadcrumb-item"
    >
      <a
        class=""
        href="/"
        target="_self"
      >
        <span>
          
        </span>
         
        <!---->
      </a>
    </li>
  </ol>
</div>
`;
